web-ui/src/main/resources/catalog/components/viewer/wfsfilter/WfsFilterDirective.js

```diff
@@ -712,15 +816,31 @@
            * Only available for administrators.
            */
           scope.indexWFSFeatures = function(version) {
-            appProfilePromise.then(function() {
-              wfsFilterService.indexWFSFeatures(
-                  scope.url,
-                  ftName,
-                  appProfile ? appProfile.tokenizedFields : null,
-                  appProfile ? appProfile.treeFields : null,
-                  uuid,
-                  version);
-            });
+            var applicationProfile = scope.md.linksTree.map(function (d) {
```

Comment:
Maybe we should continue to support both approach. md and Linkstree field may not be defined. Will rework it with a promise.

Reply:
Restored a promise which resolves to linksTree if defined.